tests/test_apply_transform_to_points.py

```diff
@@ -0,0 +1,121 @@
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import unittest
+
+import torch
+from parameterized import parameterized
+
+from monai.data import MetaTensor
+from monai.transforms.utility.array import ApplyTransformToPoints
+from monai.utils import set_determinism
+
+set_determinism(seed=0)
+
+DATA_2D = torch.rand(1, 64, 64)
+DATA_3D = torch.rand(1, 64, 64, 64)
+POINT_2D_WORLD = torch.tensor([[[2, 2], [2, 4], [4, 6]]])
+POINT_2D_IMAGE = torch.tensor([[[1, 1], [1, 2], [2, 3]]])
+POINT_2D_IMAGE_RAS = torch.tensor([[[-1, -1], [-1, -2], [-2, -3]]])
+POINT_3D_WORLD = torch.tensor([[[2, 4, 6], [8, 10, 12]], [[14, 16, 18], [20, 22, 24]]])
+POINT_3D_IMAGE = torch.tensor([[[-8, 8, 6], [-2, 14, 12]], [[4, 20, 18], [10, 26, 24]]])
+POINT_3D_IMAGE_RAS = torch.tensor([[[-12, 0, 6], [-18, -6, 12]], [[-24, -12, 18], [-30, -18, 24]]])
+
+TEST_CASES = [
+    [
+        MetaTensor(DATA_2D, affine=torch.tensor([[2, 0, 0, 0], [0, 2, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]])),
+        POINT_2D_WORLD,
+        None,
+        True,
+        False,
+        POINT_2D_IMAGE,
+    ],
+    [
+        None,
+        MetaTensor(POINT_2D_IMAGE, affine=torch.tensor([[2, 0, 0, 0], [0, 2, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]])),
+        None,
+        False,
+        False,
+        POINT_2D_WORLD,
+    ],
+    [
+        None,
+        MetaTensor(POINT_2D_IMAGE, affine=torch.tensor([[2, 0, 0, 0], [0, 2, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]])),
+        torch.tensor([[2, 0, 0, 0], [0, 2, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]]),
+        False,
+        False,
+        POINT_2D_WORLD,
+    ],
+    [
+        MetaTensor(DATA_2D, affine=torch.tensor([[2, 0, 0, 0], [0, 2, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]])),
+        POINT_2D_WORLD,
+        None,
+        True,
+        True,
+        POINT_2D_IMAGE_RAS,
+    ],
+    [
+        MetaTensor(DATA_3D, affine=torch.tensor([[1, 0, 0, 10], [0, 1, 0, -4], [0, 0, 1, 0], [0, 0, 0, 1]])),
+        POINT_3D_WORLD,
+        None,
+        True,
+        False,
+        POINT_3D_IMAGE,
+    ],
+    [
+        MetaTensor(DATA_3D, affine=torch.tensor([[1, 0, 0, 10], [0, 1, 0, -4], [0, 0, 1, 0], [0, 0, 0, 1]])),
+        MetaTensor(POINT_3D_IMAGE, affine=torch.tensor([[1, 0, 0, 10], [0, 1, 0, -4], [0, 0, 1, 0], [0, 0, 0, 1]])),
+        None,
+        False,
+        False,
+        POINT_3D_WORLD,
+    ],
+    [
+        MetaTensor(DATA_3D, affine=torch.tensor([[1, 0, 0, 10], [0, 1, 0, -4], [0, 0, 1, 0], [0, 0, 0, 1]])),
+        POINT_3D_WORLD,
+        None,
+        True,
+        True,
+        POINT_3D_IMAGE_RAS,
+    ],
+]
```

Comment:
This might format into fewer lines if you define the `affine` value as a separate variable above. You're using the same affine matrices in multiple places anyway.

Reply:
Updated it in the latest commit.